iceoryx_hoofs/include/iceoryx_hoofs/cxx/variant.hpp

```diff
@@ -265,10 +267,13 @@ class variant
     constexpr uint64_t index() const noexcept;
 
   private:
-    // NOLINTJUSTIFICATION safe access is guaranteed since the c-array is wrapped inside the variant class
-    // NOLINTBEGIN(cppcoreguidelines-avoid-c-arrays,hicpp-avoid-c-arrays)
-    alignas(algorithm::maxVal(alignof(Types)...)) internal::byte_t m_storage[TYPE_SIZE]{0U};
-    // NOLINTEND(cppcoreguidelines-avoid-c-arrays,hicpp-avoid-c-arrays)
+    struct alignas(TYPE_ALIGNMENT) storage_t
```

Comment:
If the `TYPE_ALIGNMENT` isn't used anywhere else, we could get rid of `algorithm::maxVal(alignof(Types)...);` and just use `alignas(Types...)`